### PR TITLE
Transpiler modify to accept argument from `pro`

### PR DESCRIPTION
**Pull request by ttodua**
I've amended the method to accept argument, in case it is called from `pro`, because it has its distinctive location/filenames to be written to.

### CODE REVIEW

File: build/transpile.js [hunk 1]
```diff
@@ -1355,7 +1357,7 @@ class Transpiler {
 
         const tsBodyIntellisense = tsBaseError + '\n\n    ' + tsErrors.join ('\n    ') + '\n\n'
 
-        const tsFilename = './ccxt.d.ts'
+        const tsFilename = isPro ? './ccxt.pro.d.ts' : './ccxt.d.ts'
```
Comment (kroitor) [inline build/transpile.js:1360]: Can we do this in CCXT Pro instead of CCXT ?

Comment (ttodua) [reply build/transpile.js:1360]: should we copy the whole method into pro?

File: build/transpile.js [hunk 2]
```diff
@@ -1306,9 +1308,9 @@ class Transpiler {
 
         const python3BodyIntellisense = python3Body + '\n\n\n' + pythonBaseError + '\n' + pythonErrors.join ('\n') + '\n' + pythonAll + '\n'
 
-        const pythonFilename = './python/ccxt/base/errors.py'
+        const pythonFilename = './python/ccxt' + (isPro ? 'pro' : '') + '/base/errors.py'
```
Comment (kroitor) [inline build/transpile.js:1311]: We don't have this file in CCXT Pro.

File: build/transpile.js [hunk 3]
```diff
@@ -1333,11 +1335,11 @@ class Transpiler {
 
         const phpErrors = intellisense (errorHierarchy, 'Exception', phpMakeErrorClassFile)
         const phpBodyIntellisense = phpErrors.join ("\n") + "\n\n"
-        const phpFilename = './ccxt.php'
+        const phpFilename ='./ccxt' + (isPro ? 'pro' : '') + '.php'
```
Comment (kroitor) [inline build/transpile.js:1338]: We don't have this file in CCXT Pro.

File: build/transpile.js [hunk 4]
```diff
@@ -1249,12 +1249,14 @@ class Transpiler {
...
     // ========================================================================
 
-    transpileErrorHierarchy () {
+    transpileErrorHierarchy (isPro = false, targetCcxtBaseDir = undefined) {
+
+        targetCcxtBaseDir = targetCcxtBaseDir !== undefined ? targetCcxtBaseDir : '';
 
         const errorHierarchyFilename = './js/base/errorHierarchy.js'
-        const errorHierarchy = require ('.' + errorHierarchyFilename)
+        const errorHierarchy = require (__dirname + '/.' + errorHierarchyFilename)
```
Comment (kroitor) [inline build/transpile.js:1257]: We don't have the errorHierarchy.js in CCXT Pro, let's revert this change.

File: build/transpile.js (hunk 2 above)
Comment (ttodua) [reply build/transpile.js]: ah, let me recheck

File: build/transpile.js (hunk 1 above)
Comment (ttodua) [reply build/transpile.js]: isn't this? https://github.com/kroitor/ccxt.pro/blob/master/ccxtpro.php
